### ISSUE TITLE
NVDA announcing blank on many occasions when pressing arrow keys doesn't change focus

### ISSUE THREAD

**Issue by LeonarddeR** [musescore/MuseScore#20732]
### Issue type

Accessibility issues

### Steps to reproduce

1. Open a score
2. Ensure cursor is at the start of the score
3. Press left arrow
4. NVDA says "Blank"

#### Alternative steps
1. From the score view, tab to the 'add instruments' button
2. Press left arrow
3. NVDA says "Blank"

### MuseScore Version

MuseScore 4.2

### Regression

Choose option...

### Operating system

Windows 11 23H2

### Additional context

The technical cause of this issue is actually pretty obvious. It looks like the UIA Text Pattern is implemented for many if not all controls in MuseScore. This makes NVDA think that the control contains editable text, and therefore it queries the text pattern for text when pressing arrow keys.
Instead, the text pattern should only be implemented for controls that offer text in the Text Pattern.
We can of course fix this in NVDA by ignoring the text pattern for control types other than Editable Text or document, but that's slightly hacky and I'd try to avoid thatwhen possible.

**Comment by jmwmusic** [comment]
This issue may need closing as the behaviour is now different since 4.6 and is covered by the new issue 30463

**Comment by LeonarddeR** [comment]
The underlying defect described here was never fixed, and this issue cannot be reopened by me. I have re-filed it as #34877, with the root cause and a proposed fix documented in musescore/muse_framework#284.
